docs/DesignDocs/AttestationAPIsWithFlexibleClaims.md

```diff
@@ -0,0 +1,478 @@
+Similar to the run-time claims support, the API appends the `runtime_custom_claims_buffer` to the
+`public_key` and passes the new buffer to the `oe_get_evidence` API.
+
+```diff
+oe_result_t oe_get_attestation_certificate_with_evidence_v3(
+    const oe_uuid_t* format_id,
+    const unsigned char* subject_name,
+    const uint8_t* private_key,
+    size_t private_key_size,
+    const uint8_t* public_key,
+    size_t public_key_size,
++   const uint8_t* runtime_custom_claims_buffer,
++   size_t runtime_custom_claims_buffer_size,
++   const uint8_t* inittime_custom_claims_buffer,
++   size_t inittime_custom_claims_buffer_size,
+    const void* optional_parameters,
+    size_t optional_parameters_size,
+    uint8_t** output_certificate,
+    size_t* output_certificate_size)
+{
+    oe_result_t result = OE_FAILURE;
+    uint8_t* evidence_buffer = NULL;
+    size_t evidence_buffer_size = 0;
++   uint8_t* custom_claims_buffer = NULL;
++   size_t custom_claims_buffer_size = 0;
++   uint8_t* evidence_with_inittime_custom_claims = NULL;
++   size_t evidence_with_inittime_custom_claims_size = 0;
+
+    OE_TRACE_VERBOSE("Calling oe_get_attestation_certificate_with_evidence_v3");
+    OE_TRACE_VERBOSE(
+        "generate evidence with hash from public_key_size=%d public_key key "
+        "=\n[%s]\n",
+        public_key_size,
+        public_key);
+
++   custom_claims_buffer_size = public_key_size + runtime_custom_claims_buffer_size;
++   custom_claims_buffer = oe_malloc(custom_claims_buffer_size);
++   if (!custom_claims_buffer)
++       OE_RAISE(OE_OUT_OF_MEMORY);
+
++   memcpy(custom_claims_buffer, public_key, public_key_size);
++   memcpy(
++       custom_claims_buffer + public_key_size,
++       runtime_custom_claims_buffer,
++       runtime_custom_claims_buffer_size);
+
+    result = oe_get_evidence(
+        format_id,
+        OE_EVIDENCE_FLAGS_EMBED_FORMAT_ID,
+-       public_key,
+-       public_key_size,
++       custom_claims_buffer,
++       custom_claims_buffer_size,
+        optional_parameters,
+        optional_parameters_size,
+        &evidence_buffer,
+        &evidence_buffer_size,
+        NULL,
+        0);
+    OE_CHECK_MSG(
+        result, "oe_get_evidence failed with %s\n", oe_result_str(result));
+
++   evidence_with_inittime_custom_claims_size =
++       evidence_buffer_size + inittime_custom_claims_buffer_size;
++   evidence_with_inittime_custom_claims =
++       oe_malloc(evidence_with_inittime_custom_claims_size);
++   if (!evidence_with_inittime_custom_claims)
++       OE_RAISE(OE_OUT_OF_MEMORY);
+
++   memcpy(
++       evidence_with_inittime_custom_claims, evidence_buffer, evidence_buffer_size);
++   memcpy(
++       evidence_with_inittime_custom_claims + evidence_buffer_size,
++       inittime_custom_claims_buffer,
++       inittime_custom_claims_buffer_size);
+
+    result = generate_x509_self_signed_certificate(
+        oid_oe_evidence,
+        sizeof(oid_oe_evidence),
```

Comment:
Watch out for the length limit of the generated certificate. I think we already went to mbedtls once to ask them to increase the length limit with the self signed certificate generated by OE. Now adding inittime and runtime claims definitely will increase the length even further.